api/service/staking/service.go

```diff
@@ -1,25 +1,37 @@
 package staking
 
 import (
+	"crypto/ecdsa"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/ethereum/go-ethereum/log"
+	client "github.com/harmony-one/harmony/api/client/service"
+	proto "github.com/harmony-one/harmony/api/client/service/proto"
 	"github.com/harmony-one/harmony/api/proto/message"
+	"github.com/harmony-one/harmony/core/types"
 	"github.com/harmony-one/harmony/internal/utils"
 	"github.com/harmony-one/harmony/p2p"
 )
 
 // Service is the staking service.
+// Service requires private key here which is not a right design.
+// In stead in the right design, the end-user who runs mining needs to provide signed tx to this service.
 type Service struct {
 	stopChan    chan struct{}
```

Comment:
I think there maybe away to go around the need for exposing private key, if we make it 2 steps. 
We first ask for `nonce` (via publicKey/Address) then we send `txn`. So we need not expose `privateKey`. What do you think?

Reply:
One way to do that is: getting accountState could be a independent service? It might be required elsewhere as well.

Reply:
i will add a library for this.